functional-tests/appwrapper_reconciler_test.go

```diff
@@ -0,0 +1,221 @@
+package functional_tests
+
+import (
+	"context"
+	"go/build"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	gstruct "github.com/onsi/gomega/gstruct"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	machinev1 "github.com/openshift/client-go/machine/clientset/versioned"
+	. "github.com/project-codeflare/codeflare-common/support"
+	"github.com/project-codeflare/instascale/controllers"
+	"github.com/project-codeflare/instascale/pkg/config"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	mc "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/client/clientset/versioned"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client // You'll be using this client in your tests.
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+	err       error
+)
+
+func startEnvTest(t *testing.T) *envtest.Environment {
+	test := With(t)
+	//specify testEnv configuration
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "project-codeflare", "multi-cluster-app-dispatcher@v1.38.1", "config", "crd", "bases"),
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "openshift", "api@v0.0.0-20220411210816-c3bb724c282a", "machine", "v1beta1"),
```

Comment:
The tests are failing due to this line: 
```
<*fmt.wrapError | 0xc0003fc840>: 
            unable to install CRDs onto control plane: unable to read CRD files: stat /home/runner/go/pkg/mod/github.com/openshift/api@v0.0.0-20220411210816-c3bb724c282a/machine/v1beta1: no such file or directory
            {
                msg: "unable to install CRDs onto control plane: unable to read CRD files: stat /home/runner/go/pkg/mod/github.com/openshift/api@v0.0.0-20220411210816-c3bb724c282a/machine/v1beta1: no such file or directory",
                err: <*fmt.wrapError | 0xc0003fc820>{
                    msg: "unable to read CRD files: stat /home/runner/go/pkg/mod/github.com/openshift/api@v0.0.0-20220411210816-c3bb724c282a/machine/v1beta1: no such file or directory",
                    err: <*fs.PathError | 0xc000493500>{
                        Op: "stat",
                        Path: "/home/runner/go/pkg/mod/github.com/openshift/api@v0.0.0-20220411210816-c3bb724c282a/machine/v1beta1",
                        Err: <syscall.Errno>0x2,
                    },
                },
            }
```

As this crds have a different file path:
```
go: downloading github.com/openshift/api v0.0.0-20230213134911-7ba313770556
```

resulting in:
```go
			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "openshift", "api v0.0.0-20230213134911-7ba313770556", "machine", "v1beta1"),

```

Reply:
IMHO they are derived from `go.mod`
I think we can keep it to be updated manually for now, if it will become a burden then some automation can be developed.